cmd/testscript/main.go

```diff
@@ -85,14 +84,26 @@ func mainerr() (retErr error) {
 		files = []string{"-"}
 	}
 
-	for i, fileName := range files {
+	dirNames := make(map[string]int)
+	for _, filename := range files {
 		// TODO make running files concurrent by default? If we do, note we'll need to do
 		// something smarter with the runner stdout and stderr below
-		runDir := filepath.Join(td, strconv.Itoa(i))
+
+		// Derive a name for the directory from the basename of file, making
+		// uniq by adding a numeric suffix in the case we otherwise end
+		// up with multiple files with the same basename
+		dirName := filepath.Base(filename)
+		count := dirNames[dirName]
+		dirNames[dirName] = count + 1
+		if count != 0 {
+			dirName = fmt.Sprintf("%v%v", dirName, count)
```

Comment:
nit: use `%s%d` as it's less ambiguous, and vet will check the types for you.

Reply:
interesting that the rest of the file uses `%v` too. I personally only ever use that for interfaces or complex types like maps and structs.

Reply:
Good point, thanks. Muscle memory more than anything.